bin/ocFunctions.inc

```diff
@@ -1788,17 +1788,69 @@ generateSeed() {
 function getSecret() {
   (
     resourceName=${1}
-    key=${2}
+    keys=${2}
     projectName=${3:-$(getProjectName)}
-    if [ -z "${resourceName}" ] || [ -z "${key}" ]; then
+    if [ -z "${resourceName}" ] || [ -z "${keys}" ]; then
       echoError "\ngetSecret; You must specify the 'resourceName' and 'key'.\n"
       exit 1
     fi
 
-    echo $(oc -n ${projectName} extract --to=- --keys=${key} secret/${resourceName} 2>&1 | sed -n 2p)
+    echo $(oc -n ${projectName} extract --to=- --keys=${keys} secret/${resourceName} 2>&1 | sed -n '/#/ !p')
   )
 }
 
+function buildPromptCreds(){
+  namespaceName=${1}
+  newCredName=${2}
+  dockerReg=${3}
+  credEnvs=${4}
+  userName=${5}
+  password=${6}
+  
+  if ([ -z ${userName} ] || [ -z ${password} ]); then
+    echo "Please enter your ${dockerReg} username"
+    read userName
+
+    echo "Please enter your ${dockerReg} password"
+    read password
+  fi
+
+  #set cred env to tools if unset
+  if [ -z "${credEnvs}" ]; then
+    credEnvs="tools"
+  fi
+  #create pull secret in each environment we need
+  for env in ${credEnvs}; do
+    oc create secret docker-registry ${newCredName} -n ${namespaceName}-${env} \
+      --docker-server=${dockerReg} \
+      --docker-username=${userName} \
+      --docker-password=${password} \
+      --docker-email=unused
```

Comment:
Not sure it makes a difference, but I have been using what the Artifactory docs highlight for the email, which is `--docker-email=<username>@<namespace>.local` - we could stick to that.